include/pros/optical.h

```diff
@@ -0,0 +1,271 @@
+/**
+ * \file pros/optical.h
+ *
+ * Contains prototypes for functions related to the VEX Optical sensor.
+ *
+ * Visit https://pros.cs.purdue.edu/v5/tutorials/topical/imu.html to learn
+ * more.
+ *
+ * This file should not be modified by users, since it gets replaced whenever
+ * a kernel upgrade occurs.
+ *
+ * Copyright (c) 2017-2020, Purdue University ACM SIGBots.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#ifndef _PROS_OPTICAL_H_
+#define _PROS_OPTICAL_H_
+
+#include <stdbool.h>
+#include <stdint.h>
+
+#define OPT_GESTURE_ERR (INT8_MAX)
+#define OPT_COUNT_ERR (INT16_MAX)
+#define OPT_TIME_ERR PROS_ERR
+
+#ifdef __cplusplus
+extern "C" {
+namespace pros {
+namespace c {
+#endif
+
+typedef enum optical_direction_e {
+	NO_GESTURE = 0,
+	UP = 1,
+	DOWN = 2,
+	RIGHT = 3,
+	LEFT = 4,
+} optical_direction_e_t;
```

Comment:
consider making these lower case. All upper case is usually reserved for macros.